src/facade/redis_parser.cc

```diff
@@ -21,36 +21,34 @@ auto RedisParser::Parse(Buffer str, uint32_t* consumed, RespExpr::Vec* res) -> R
   }
 
   if (state_ == CMD_COMPLETE_S) {
-    state_ = INIT_S;
-  }
-
-  if (state_ == INIT_S) {
     InitStart(str[0], res);
   }
+  DCHECK(state_ != CMD_COMPLETE_S);
```

Comment:
I had to check this. `InitStart` does update `state_` if it's equal `CMD_COMPLETE_S` so it's fine.